packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -264,6 +262,22 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         this.bindState = existing ? BindState.Bound : BindState.NotBound;
         this._attachState = dataStoreContext.attachState;
 
+        /**
+         * back-compat 0.58.2000 - visibility state was added in this version. For older versions, get the visibility
+         * state similar to the data store context:
+         * The existing flag can be true in two conditions:
+         * 1. It's a local data store that is created when a detached container is rehydrated. In this case, the data
+         *    store is locally visible because the snapshot it is loaded from contains locally visible data stores only.
+         * 2. It's a remote data store that is created when an attached container is loaded is loaded from snapshot or
+         *    when an attach op comes in. In both these cases, the data store is already globally visible.
+         */
+        if (dataStoreContext.visibilityState !== undefined) {
```

Comment:
Correct me if I got it wrong, but I believe we have two places where we track same thing - here and in context.
And that might be source of mismatches / bugs.
Should this be a calculated property, that always relies on state of context (basically this logic moved to getter and this.visibilityState is removed)?

Reply:
It would have to be the other way around - Have the state here and context gets it from here. The reason is that this state propagates from the runtime to the context compared to something like connectivity state which propagates the other way around.
The runtime has to update all its dependencies before it can let the channel know of the new state so it has to update the state locally first.

Reply:
Sure, should we do so (tracking in one place)? That probably can be done only in 2 releases, I guess.

Reply:
Data store context doesn't need to track visibility state because it doesn't use it. I removed it from the context and added it to `IFluidDataStoreChannel` so that the context can access it if needed.